proxy/cache.go

```diff
@@ -39,12 +39,12 @@ func (c *cache) Get(request *dns.Msg) (*dns.Msg, bool) {
 		return nil, false
 	}
 
-	res := unpackResponse(data, request)
+	res, isNotExpired := unpackResponse(data, request)
 	if res == nil {
-		c.items.Del(key)
+		//c.items.Del(key)	no need to delete it
```

Comment:
It's better to not leave commented-out code.

Reply:
fixed